.github/workflows/update-robots-txt.yml

```diff
@@ -0,0 +1,88 @@
+name: Update robots.txt
+
+permissions: write-all
+
+on: 
+  schedule:
+    - cron: '0 0 * * 0'
+  workflow_dispatch:
+
+jobs:
+  update-robots:
+    runs-on: ubuntu-latest
+    steps: 
+      - name: Create Branch
+        env:
+          GITHUB_TOKEN: ${{ github.token }}
+        uses: peterjgrainger/action-create-branch@v2.2.0
+        with:
+          branch: robots.txt-update
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          ref: robots.txt-update
+          fetch-depth: 0 
+      - name: Update robot.txt
+        id: update
+        env:
+          API_KEY: ${{ secrets.ROBOTS }}
+        run: |
+          '#       _---~~(~~-_.
+          #     _{        )   )
+          #   ,   ) -~~- ( ,-' )_
+          #  (  `-,_..`., )-- '_,)
+          # ( ` _)  (  -~( -_ `,  }
+          # (_-  _  ~_-~~~~`,  ,' )
+          #   `~ -^(    __;-,((()))
+          #         ~~~~ {_ -_(())
+          #                `\  }
+          #                  { }  
+          # BRAAAAAAAAIIIINNSSSSSSS
+
+          ' >> public/robots.txt
+          curl --location 'https://api.darkvisitors.com/robots-txts' \
+          --header 'Content-Type: application/json' \
+          --header 'Authorization: Bearer $API_KEY' \
+          --data '{ "agent_types": [ "AI Data Scraper", "AI Assistant", "AI Search Crawler" ], "disallow": "/" }' > public/robots.txt
+          git add public/robots.txt
```

Comment:
Did u see https://github.com/marketplace/actions/create-pull-request ? 
I think it handles the diffing and deduping nicely, eg  we used it here https://github.com/linkml/linkml/blob/03b9003d876ef179e26d2de6bc8f711e3de4d1eb/.github/workflows/poetry-update.yaml#L55
Where we just modify the file in one step and it handles the rest.

If this works then all good with me, but just dont want u to spend too much time if we end up needing to debug stuff :)

Reply:
oh yeah, that's way nicer :)